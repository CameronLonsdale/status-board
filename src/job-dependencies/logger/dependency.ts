import * as Tracer from 'tracer';
import logger from '../../logger';

export default function (jobWorker: any, deps: any): Tracer.Logger {
  return logger(jobWorker, deps.io);
}
