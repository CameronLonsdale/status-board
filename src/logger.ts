import { Console } from 'console';
import configManager from './config-manager';

const config = configManager('logging');
export const logger = new Console(process.stdout, process.stderr);
export function formatConsoleDate(date: Date) {
  const hour = date.getHours();
  const minutes = date.getMinutes();
  const seconds = date.getSeconds();
  const milliseconds = date.getMilliseconds();

  return '[' +
    ((hour < 10) ? '0' + hour : hour) +
    ':' +
    ((minutes < 10) ? '0' + minutes : minutes) +
    ':' +
    ((seconds < 10) ? '0' + seconds : seconds) +
    '.' +
    ('00' + milliseconds).slice(-3) +
    ']: ';
}

export default function (jobWorker?: any, io?: SocketIO.Server) {
  // jobWorker and socket.io instance are optional
  const loggerConfig = config.logger || {};
  // tslint:disable-next-line max-line-length
  const prefix = jobWorker ? `[dashboard: ${jobWorker.dashboard_name}] [job: ${jobWorker.job_name}] ` : '';

  loggerConfig.transport = (data: any) => {
    const logText = prefix + data.output;
    // tslint:disable-next-line no-console
    console.log(logText);
    if (io) {
      io.emit('server', { type: data.level, msg: logText });
    }
  };

  return logger;
}
