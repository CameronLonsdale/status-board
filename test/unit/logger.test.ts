import logger from '../../src/logger';

describe('Logger', () => {
  test('Should match snapshot', () => {
    expect(logger).toMatchSnapshot();
  });
});
