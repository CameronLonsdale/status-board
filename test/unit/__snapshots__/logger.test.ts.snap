// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logger Should match snapshot 1`] = `[Function]`;
